<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>652</width>
    <height>700</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Generar clase Storm + .ui</string>
  </property>
  <property name="styleSheet">
   <string notr="true">QComboBox {
border: 1px solid #999999; 
border-radius: 4px;
padding: 3px 2px 2px 2px;
min-width: 6em;
}
QComboBox::down-arrow {image: url(:newPrefix/arrow_down.png);height: 10px;}
QComboBox:editable {background: white;border-radius: 0px;}
QComboBox::drop-down { border-style: none}

QLineEdit, QDoubleSpinBox, QTextEdit  { 
border: 1px solid #999999; 
padding: 2px 2px 2px 2px;
min-width: 90px;
height: 20px;

border-top-left-radius: 3px ;
border-top-right-radius: 3px;
border-bottom-right-radius: 3px;
border-bottom-left-radius: 3px;
}

QLineEdit:focus, QDoubleSpinBox:focus, QTextEdit:focus{ border: 1px solid rgb(0, 188, 0);}</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout_2">
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_14">
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_11">
        <item>
         <widget class="QLabel" name="label_7">
          <property name="font">
           <font>
            <weight>75</weight>
            <bold>true</bold>
           </font>
          </property>
          <property name="text">
           <string>Nombre Clase</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="leNombreClase"/>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_12">
        <item>
         <widget class="QLabel" name="label_8">
          <property name="font">
           <font>
            <weight>75</weight>
            <bold>true</bold>
           </font>
          </property>
          <property name="text">
           <string>Base de datos</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QComboBox" name="cbBDs">
          <item>
           <property name="text">
            <string>SQLite</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>My SQL</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>PostgreSQL</string>
           </property>
          </item>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </item>
    <item>
     <widget class="QGroupBox" name="groupBox">
      <property name="font">
       <font>
        <weight>75</weight>
        <bold>true</bold>
       </font>
      </property>
      <property name="title">
       <string>Agregar atributo</string>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_4">
       <item>
        <layout class="QVBoxLayout" name="verticalLayout">
         <property name="leftMargin">
          <number>120</number>
         </property>
         <property name="rightMargin">
          <number>120</number>
         </property>
         <property name="bottomMargin">
          <number>0</number>
         </property>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout">
           <item>
            <widget class="QLabel" name="label">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="text">
              <string>Nombre atributo</string>
             </property>
             <property name="alignment">
              <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLineEdit" name="leNombre">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="styleSheet">
              <string notr="true">background-color: rgb(255, 255, 255);</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_4">
           <item>
            <widget class="QLabel" name="label_4">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="text">
              <string>Widget</string>
             </property>
             <property name="alignment">
              <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QComboBox" name="cbWidget">
             <property name="styleSheet">
              <string notr="true">background-color: rgb(255, 255, 255);</string>
             </property>
             <item>
              <property name="text">
               <string>QLineEdit</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>QComboBox</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>QRadioButton</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>QSpinBox</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>QDateEdit</string>
              </property>
             </item>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_2">
           <item>
            <widget class="QLabel" name="label_2">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="text">
              <string>Storm Type</string>
             </property>
             <property name="alignment">
              <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QComboBox" name="cbStormType">
             <property name="styleSheet">
              <string notr="true">background-color: rgb(255, 255, 255);</string>
             </property>
             <property name="currentIndex">
              <number>11</number>
             </property>
             <property name="insertPolicy">
              <enum>QComboBox::InsertAlphabetically</enum>
             </property>
             <item>
              <property name="text">
               <string>Bool</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>Date</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>DateTime</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>Decimal</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>Float</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>Int</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>List</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>Pickle</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>RawStr</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>Time</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>TimeDelta</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>Unicode</string>
              </property>
             </item>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_3">
           <item>
            <widget class="QLabel" name="label_3">
             <property name="text">
              <string>Valor por defecto</string>
             </property>
             <property name="alignment">
              <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLineEdit" name="leValorPorDefecto">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="styleSheet">
              <string notr="true">background-color: rgb(255, 255, 255);</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_10">
           <item>
            <widget class="QPushButton" name="btPrimario">
             <property name="text">
              <string>Primario?</string>
             </property>
             <property name="checkable">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="btNotNull">
             <property name="text">
              <string>Not Null?</string>
             </property>
             <property name="checkable">
              <bool>true</bool>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QGroupBox" name="gbReferencia">
         <property name="title">
          <string>Referencia</string>
         </property>
         <property name="checkable">
          <bool>true</bool>
         </property>
         <property name="checked">
          <bool>false</bool>
         </property>
         <layout class="QHBoxLayout" name="horizontalLayout_9">
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_6">
            <item>
             <widget class="QLabel" name="label_5">
              <property name="text">
               <string>Nombre clase</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLineEdit" name="leNombreReferencia">
              <property name="styleSheet">
               <string notr="true">background-color: rgb(255, 255, 255);</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <widget class="QCheckBox" name="chkCruzada">
            <property name="text">
             <string>Cruzada?</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="btAgregar">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="maximumSize">
          <size>
           <width>1666666</width>
           <height>1666666</height>
          </size>
         </property>
         <property name="styleSheet">
          <string notr="true">QPushButton {
	background: #DBDBDB;
	border-radius: 3px;
	color:#353535;
	border: 1px solid #A0A197;
	padding:8px;
}

QPushButton:pressed{
	background-color: qlineargradient(x1: 0, y1: 0, x2: 0, y2: 1,
                                            stop: 0 #dadbde, stop: 1 #f6f7fa);		
	border: 1px solid #505050;
}
QPushButton:hover{
         background-color: qlineargradient(x1: 0, y1: 0, x2: 0, y2: 1,
                                           stop: 0 #dadbde, stop: 1 #f6f7fa);
}</string>
         </property>
         <property name="text">
          <string>+ Agregar</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="Line" name="line">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_8">
      <item>
       <widget class="QTableWidget" name="twLista">
        <attribute name="horizontalHeaderCascadingSectionResizes">
         <bool>true</bool>
        </attribute>
        <attribute name="horizontalHeaderDefaultSectionSize">
         <number>150</number>
        </attribute>
        <attribute name="horizontalHeaderStretchLastSection">
         <bool>true</bool>
        </attribute>
        <attribute name="horizontalHeaderStretchLastSection">
         <bool>true</bool>
        </attribute>
        <attribute name="horizontalHeaderCascadingSectionResizes">
         <bool>true</bool>
        </attribute>
        <attribute name="horizontalHeaderDefaultSectionSize">
         <number>150</number>
        </attribute>
        <column>
         <property name="text">
          <string>Nombre atributo</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>Storm Type</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>Widget</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>Valor por defecto</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>Primario</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>Not Null</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>Referencia</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>Cruzada</string>
         </property>
        </column>
       </widget>
      </item>
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_3">
        <item>
         <widget class="QPushButton" name="btQuitar">
          <property name="text">
           <string>Quitar</string>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="verticalSpacer">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </item>
     </layout>
    </item>
    <item>
     <widget class="Line" name="line_2">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_13">
      <item>
       <widget class="QCheckBox" name="chkGenerarClase">
        <property name="text">
         <string>Generar clase Storm.</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QCheckBox" name="chkGenerarUi">
        <property name="text">
         <string>Generar .ui para esta clase</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QCheckBox" name="chkGenerarPaquete">
        <property name="text">
         <string>Generar paquete para este objeto.</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_7">
      <item>
       <widget class="QLabel" name="label_6">
        <property name="text">
         <string>Ubicación</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="leUbicacion"/>
      </item>
      <item>
       <widget class="QPushButton" name="btExaminar">
        <property name="text">
         <string>Examinar...</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_5">
      <item>
       <spacer name="horizontalSpacer">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QPushButton" name="btSalir">
        <property name="text">
         <string>Salir</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="btGenerar">
        <property name="cursor">
         <cursorShape>PointingHandCursor</cursorShape>
        </property>
        <property name="styleSheet">
         <string notr="true">QPushButton{
	color: rgb(255, 255, 255);
	background-color: qlineargradient(spread:pad, x1:0.005, y1:0.006, x2:0.990476, y2:1, stop:0.0047619  rgba(10, 134, 2, 255), stop:1 rgba(6, 84, 0, 255));
border: 1px solid #505050;
border-radius: 3px;
padding:6px 20px;
font: bold;
}
QPushButton:pressed{	
	border: 3px solid #505050;
}</string>
        </property>
        <property name="text">
         <string>Generar</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
 </widget>
 <tabstops>
  <tabstop>leNombreClase</tabstop>
  <tabstop>cbBDs</tabstop>
  <tabstop>leNombre</tabstop>
  <tabstop>cbStormType</tabstop>
  <tabstop>cbWidget</tabstop>
  <tabstop>leValorPorDefecto</tabstop>
  <tabstop>gbReferencia</tabstop>
  <tabstop>leNombreReferencia</tabstop>
  <tabstop>chkCruzada</tabstop>
  <tabstop>btAgregar</tabstop>
  <tabstop>twLista</tabstop>
  <tabstop>btQuitar</tabstop>
  <tabstop>chkGenerarUi</tabstop>
  <tabstop>btExaminar</tabstop>
  <tabstop>leUbicacion</tabstop>
  <tabstop>btGenerar</tabstop>
  <tabstop>btSalir</tabstop>
 </tabstops>
 <resources>
  <include location="images_rc/images.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>btSalir</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>540</x>
     <y>657</y>
    </hint>
    <hint type="destinationlabel">
     <x>460</x>
     <y>567</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>leNombre</sender>
   <signal>returnPressed()</signal>
   <receiver>btAgregar</receiver>
   <slot>click()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>520</x>
     <y>97</y>
    </hint>
    <hint type="destinationlabel">
     <x>300</x>
     <y>313</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
